crates/mongodb-agent-common/src/query/serialization/json_to_bson.rs

```diff
@@ -84,37 +85,34 @@ pub fn json_to_bson_scalar(expected_type: BsonScalarType, value: Value) -> Resul
                 )
             })?,
         ),
-        BsonScalarType::String => Bson::String(deserialize(expected_type, value)?),
-        BsonScalarType::Date => convert_date(&from_string(expected_type, value)?)?,
-        BsonScalarType::Timestamp => {
-            deserialize::<json_formats::Timestamp>(expected_type, value)?.into()
-        }
-        BsonScalarType::BinData => {
-            deserialize::<json_formats::BinData>(expected_type, value)?.into()
-        }
-        BsonScalarType::ObjectId => Bson::ObjectId(deserialize(expected_type, value)?),
-        BsonScalarType::Bool => match value {
+        S::String => Bson::String(deserialize(expected_type, value)?),
+        S::Date => convert_date(&from_string(expected_type, value)?)?,
+        S::Timestamp => deserialize::<json_formats::Timestamp>(expected_type, value)?.into(),
+        S::BinData => deserialize::<json_formats::BinData>(expected_type, value)?.into(),
+        S::UUID => convert_uuid(&from_string(expected_type, value)?)?,
```

Comment:
I'm sorry about the churn in this function. This is the only added line. The logic in other lines is unchanged.